tempesta_fw/sock_clnt.c

```diff
@@ -128,9 +126,11 @@ tfw_cli_conn_send(TfwCliConn *cli_conn, TfwMsg *msg)
 	int r;
 
 	r = tfw_connection_send((TfwConn *)cli_conn, msg);
-	mod_timer(&cli_conn->timer,
-		  jiffies +
-		  msecs_to_jiffies((long)tfw_cli_cfg_ka_timeout * 1000));
+
+	if (timer_pending(&cli_conn->timer))
+		mod_timer(&cli_conn->timer,
```

Comment:
Generally speaking, this check is unreliable: the timer can be deactivated right between the check and `mod_timer()`. Why we ever call `tfw_cli_conn_send()` on dropped connection? There are only 2 places calling the function:

1. ` tfw_http_resp_fwd()` -> `__tfw_http_resp_fwd()` - the former one does `tfw_cli_conn_get()`, so the connection is alive;

2. `tfw_h2_resp_fwd()` - this one seems doesn't care about the client connection state.

The interesting thing is that there are only two places where connection reference counter is incremented: the 1st one in HTTP/1 forwarding and the 2nd one is `tfw_http_conn_msg_alloc()`. Nobody calls `tfw_srv_conn_get()`.

So it seems HTTP/2 is prone for the this and I think other similar referencing a dead connection bugs. Please test HTTP/2 code against #1428 

There are two macros over `tfw_connection_get()`, which are not needed.